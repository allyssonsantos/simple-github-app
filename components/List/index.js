import React, { Component } from 'react';
import styled from 'styled-components';

import Item from './sub-components/Item';

const StyledUl = styled.ul`
  list-style: none;
  margin: 0;
  padding-left: 0;
`;

class List extends Component {
  constructor(props) {
    super(props);
  }

  static Item = Item;

  render() {
    const { children, ...rest } = this.props;
    return <StyledUl {...rest}>{children}</StyledUl>;
  }
}

export default List;
