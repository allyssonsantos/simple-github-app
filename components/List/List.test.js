import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import List from './';

describe('With Snapshot Testing', () => {
  it('Should match snapshot without children', () => {
    const component = renderer.create(<List />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should match snapshot with childrens', () => {
    const ListComponent = () => (
      <List>
        <List.Item>First Item</List.Item>
        <List.Item>Second Item</List.Item>
      </List>
    );
    const component = renderer.create(<ListComponent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
