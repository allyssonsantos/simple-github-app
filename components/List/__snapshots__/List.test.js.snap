// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Should match snapshot with childrens 1`] = `
.c0 {
  list-style: none;
  margin: 0;
  padding-left: 0;
}

<ul
  className="c0"
>
  <li
    className=""
  >
    First Item
  </li>
  <li
    className=""
  >
    Second Item
  </li>
</ul>
`;

exports[`With Snapshot Testing Should match snapshot without children 1`] = `
.c0 {
  list-style: none;
  margin: 0;
  padding-left: 0;
}

<ul
  className="c0"
/>
`;
